scripts/python/gen.py

```diff
@@ -230,6 +234,12 @@ def _add_cobbler_distros(self):
         print('Success: Cobbler distros and profiles added')
 
     def _inv_add_ports_pxe(self):
+        power_time_out = gen.get_power_time_out()
```

Comment:
When running gen deploy, validate_cluster_hardware has already set boot device to network and power cycled the nodes quite a bit earlier in the process.

Reply:
I was thinking that adding the power cycle here would ensure 'deploy --inv-add-ports-pxe' could run successfully no matter when it is called (e.g. if we're debugging and needing to run manually). That said, "hidden" power cycles can be a little dangerous. Now I'm thinking I could add extra arguments (maybe '--pxe-boot-nodes', or '--set-bootdev-pxe' + '--power-cycle-nodes') to accomplish the same.

Taking it out of this patch would be very easy, so if we think that makes sense let's do it now.

Reply:
I agree we do need a way to cycle power on the nodes when debugging or or if trying to re-run starting from somewhere other than the beginning.  So yes, an additional call / argument available from gen.py would be useful.  I would like to add a 'tools' or 'utils' command to gen.py.  Cycling node power would be a good candidate for that.  You could then run gen utils --power-cycle-nodes.  or gen utils --set-bootdev-pxe.  Other candidate commands would be --scan-pxe-network, --scan-ipmi-network, --login-node .....